packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/stoppable_timer.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+
+/// {@template aws_logging_cloudwatch.stoppable_timer}
+/// A Timer that can be stopped and started again.
+///
+/// The `callback` function is invoked repeatedly with [duration] intervals
+/// until stopped with the [stop] function.
+/// {@endtemplate}
+class StoppableTimer {
+  /// {@macro aws_logging_cloudwatch.stoppable_timer}
+  StoppableTimer({
+    required this.duration,
+    required Future<void> Function() callback,
+    required void Function(Object) onError,
+  })  : _callback = callback,
+        _timer = Timer.periodic(duration, (Timer t) {
+          callback().catchError((Object e) {
```

Comment:
the callback function in the logger plugin handles multiple calls. if a sync is in progress it returns immediately. I don't think we need to queue the calls. if a sync is in progress when the timer invokes the callback function it doesn't need to queue the call and sync again after the pervious sync is finished. it can wait for the next interval to sync. wdyt? am I missing something here?